src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php

```diff
@@ -409,4 +411,108 @@ public function supportsDescendantType(string $type): bool
     {
         return $this->getClassName() === $type;
     }
+
+    private function createChildCollectionsQueryBuilder(string $alias, int $rootCollectionId): QueryBuilder
+    {
+        $rootCollectionAlias = $alias . '_rootCollection';
+
+        return $this->createQueryBuilder($alias)
+            ->innerJoin(
+                CollectionInterface::class,
+                $rootCollectionAlias,
+                Join::WITH,
+                $alias . '.lft > ' . $rootCollectionAlias . '.lft AND ' . $alias . '.rgt < ' . $rootCollectionAlias . '.rgt'
+            )
+            ->where($rootCollectionAlias . '.id = :id')
+            ->setParameter('id', $rootCollectionId);
+    }
+
+    /**
+     * @return array<array{id: int, resourceKey: string, depth: int}>
+     */
+    public function findChildCollectionResourcesOfRootCollection(int $rootCollectionId): array
```

Comment:
i think i would use `descendant`/`ancestor`/ instead of `root`. we dont need to include `child` then and can use `findDescendantCollectionResources`/`findCollectionResourcesByAncestor`